minid_client/minid_client_api.py

```diff
@@ -46,6 +61,47 @@ def create_default_config():
     config_file.close()
 
 
+def globus_auth(login=True):
+    """
+    Login using Globus Credentials. Supports both cached tokens and fresh
+    logins.
+
+    It will exit with an error if the globus_sdk is not installed.
+
+    :param login: Do a login instead of using cached toknes
+    :return: A globus access_token for the minid server, or None
+    """
+
+    if not login:
+        config = ConfigParser()
+        config.read(DEFAULT_CONFIG_FILE)
+        try:
+            if config.get('globus', 'access_token'):
+                return config.get('globus', 'access_token')
+        except NoSectionError:
+            return None
+
+    if not GLOBUS_AUTH_ENABLED:
+        print('"globus-sdk" is not installed, you can install it with '
+              '"pip install globus-sdk"')
+        sys.exit(-1)
+
+    client = globus_sdk.NativeAppAuthClient(GLOBUS_CLIENT_ID)
+    client.oauth2_start_flow(requested_scopes=GLOBUS_SCOPES)
+    authorize_url = client.oauth2_get_authorize_url()
+    print('Visit the URL below: \n{}'.format(authorize_url))
+    auth_code = get_input('Please enter the code you '
+                          'get after login here: ').strip()
+
+    token_response = client.oauth2_exchange_code_for_tokens(auth_code)
+    tokens = token_response.by_resource_server
+    minid_at = tokens['f238f634-6df8-414d-b3ea-77f2062f0c71']['access_token']
+    print('Add the following section to {} for future requests:'
+          '\n[globus]\naccess_token = {}\n\n'
+          ''.format(DEFAULT_CONFIG_PATH, minid_at))
+    return minid_at
```

Comment:
Saving tokens to the config file will clobber any commented out lines with ConfigParser. Since this file is only written once and it appeared in other docs users were to directly edit it, I elected not to directly write to the config file. 

Refresh tokens weren't included since handling them through the SDK would have meant significant changes to how the minid client makes calls to the minid server.